src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts

```diff
@@ -123,8 +126,46 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
     this.setFocusToTop();
   }
 
+  // Adding validation message to show it as Error Summary
+  generateErrorMessage(fields: CaseField[]){   
+    fields.filter(casefield => !this.caseFieldService.isReadOnly(casefield))
+          .forEach(casefield => {
+            if((this.editForm.controls['data'].get(casefield.id).hasError('required'))){
+              this.validationErrors.push(`${casefield.label} is required #${casefield.id}`);
+              this.editForm.controls['data'].get(casefield.id).markAsTouched();    
+            } else if ((this.editForm.controls['data'].get(casefield.id).hasError('pattern'))){
+              this.validationErrors.push(`${casefield.label} is not valid #${casefield.id}`);
+              this.editForm.controls['data'].get(casefield.id).markAsTouched();              
+            } else if ((this.editForm.controls['data'].get(casefield.id).hasError('minlength'))){
+              this.validationErrors.push(`${casefield.label} required minimum length #${casefield.id}`);
+              this.editForm.controls['data'].get(casefield.id).markAsTouched();              
+            } else if ((this.editForm.controls['data'].get(casefield.id).hasError('maxlength'))){
+              this.validationErrors.push(`${casefield.label} exceeds maximum length #${casefield.id}`);
+              this.editForm.controls['data'].get(casefield.id).markAsTouched();             
+            }             
+          })
+    this.scrollToTop();
+  }
+
+  getElementId(errorMessage: string){
+    return errorMessage.substring(errorMessage.lastIndexOf("#"));
+  }
+
+  getValidationErrorMessage(errorMessage: string){
+    return errorMessage.substring(0,errorMessage.lastIndexOf("#"));
+  }
+
+  onClick(elementId: string): void{
```

Comment:
Change the name of this function to something more specific please.

Reply:
As mentioned changed the function name to navigateToErrorElement()